src/main/java/hudson/plugins/jacoco/JacocoPublisher.java

```diff
@@ -34,348 +44,445 @@
  * @author Kohsuke Kawaguchi
  * @author Jonathan Fuerth
  * @author Ognjen Bubalo
- * 
  */
-public class JacocoPublisher extends Recorder {
-    
+public class JacocoPublisher extends Recorder implements SimpleBuildStep {
+
     /**
      * Rule to be enforced. Can be null.
-     *
+     * <p>
      * TODO: define a configuration mechanism.
      */
     public Rule rule;
-    @Deprecated public transient String includes;
-    @Deprecated public transient int moduleNum;
+    @Deprecated
+    public transient String includes;
+    @Deprecated
+    public transient int moduleNum;
     /**
      * {@link hudson.model.HealthReport} thresholds to apply.
      */
     public JacocoHealthReportThresholds healthReports;
 
-    
+
     /**
      * Variables containing the configuration set by the user.
      */
-	private final String execPattern;
-	private final String classPattern;
-	private final String sourcePattern;
-	private final String inclusionPattern;
-	private final String exclusionPattern;
-	
-    
-    private final String minimumInstructionCoverage;
-    private final String minimumBranchCoverage;
-    private final String minimumComplexityCoverage;
-    private final String minimumLineCoverage;
-    private final String minimumMethodCoverage;
-    private final String minimumClassCoverage;
-    private final String maximumInstructionCoverage;
-    private final String maximumBranchCoverage;
-    private final String maximumComplexityCoverage;
-    private final String maximumLineCoverage;
-    private final String maximumMethodCoverage;
-    private final String maximumClassCoverage;
-    private final boolean changeBuildStatus;
-    
-
-	/**
+    private String execPattern;
+    private String classPattern;
+    private String sourcePattern;
+    private String inclusionPattern;
+    private String exclusionPattern;
+
+
+    private String minimumInstructionCoverage;
+    private String minimumBranchCoverage;
+    private String minimumComplexityCoverage;
+    private String minimumLineCoverage;
+    private String minimumMethodCoverage;
+    private String minimumClassCoverage;
+    private String maximumInstructionCoverage;
+    private String maximumBranchCoverage;
+    private String maximumComplexityCoverage;
+    private String maximumLineCoverage;
+    private String maximumMethodCoverage;
+    private String maximumClassCoverage;
+    private boolean changeBuildStatus;
+
+    @DataBoundConstructor
+    public JacocoPublisher() {
```

Comment:
The new constructor could just be empty since you have all the new `@DataboundSetters`. This would put an end to the sequence of `@Deprecated` constructors.